google-cloud-firestore/src/main/java/com/google/cloud/firestore/CustomClassMapper.java

```diff
@@ -630,32 +630,32 @@ private static <T> T convertBean(Object o, Class<T> clazz, DeserializeContext co
     }
   }
 
-  private static IllegalArgumentException serializeError(ErrorPath path, String reason) {
+  private static IllegalArgumentException serializeError(
+      DeserializeContext.ErrorPath path, String reason) {
     reason = "Could not serialize object. " + reason;
     if (path.getLength() > 0) {
       reason = reason + " (found in field '" + path.toString() + "')";
     }
     return new IllegalArgumentException(reason);
   }
 
-  private static RuntimeException deserializeError(ErrorPath path, String reason) {
+  private static RuntimeException deserializeError(
+      DeserializeContext.ErrorPath path, String reason) {
     reason = "Could not deserialize object. " + reason;
     if (path.getLength() > 0) {
       reason = reason + " (found in field '" + path.toString() + "')";
     }
     return new RuntimeException(reason);
   }
 
+  private static boolean isRecordType(Class<?> cls) {
+    Class<?> parent = cls.getSuperclass();
+    return parent != null && "java.lang.Record".equals(parent.getName());
+  }
+
   // Helper class to convert from maps to custom objects (Beans), and vice versa.
-  private static class BeanMapper<T> {
-    private final Class<T> clazz;
+  private static class PojoBeanMapper<T> extends BeanMapper<T> {
```

Comment:
- [ ] moved PojoBeanMapper to separate class
- [ ] move serialization related files into `serialization` folder
- [ ] "Many methods have had their private attribute removed" 
- [ ] move duplictated code to `BeanMapper`
- [ ] move `serializeError ` and `deserializeError ` to `DeserializeContext`

Reply:
> The BeanMapper class could be a better place for common logic, since methods can be `protected` here.

I agree. Having mapper subclasses call an external utility class is less modular.

Reply:
make sense